doc/user/methods/adaboost_regressor.md

```diff
@@ -0,0 +1,582 @@
+// Print the MSE of the trained tree.
+std::cout << "MSE of trained tree is " << mse << "." << std::endl;
+
+// Compute prediction on the first test point.
+const double firstPrediction = adr.Predict(testData.col(0));
+std::cout << "Predicted value for first test point is " << firstPrediction
+    << "." << std::endl;
+
+// Compute predictions on test data.
+arma::rowvec testPredictions;
+adr.Predict(testData, testPredictions);
+
+// Compute the average error on the test set.
+const double testAverageError = arma::mean(testResponses - testPredictions);
+std::cout << "Average error on test set: " << testAverageError << "."
+    << std::endl;
+```
+
+---
+
+Load a adaboost regressor and print some information about it.
+
+```c++
+mlpack::AdaBoostRegressor<> adr;
+// This call assumes a adaboost regressor called "adr" has already been saved to
+// `adr.bin` with `data::Save()`.
+mlpack::data::Load("adr.bin", "adr", adr, true);
+
+std::cout << "The adaboost regressor in 'adr.bin' contains " << adr.NumTrees()
+    << " trees." << std::endl;
+if (adr.NumTrees() > 0)
+{
+  std::cout << "The first tree's root node has " << adr.Tree(0).NumChildren()
+      << " children." << std::endl;
+}
+```
+
+---
+
+Train a adaboost regressor on categorical data, and compare its performance 
+with the performance of each individual tree:
+
+```c++
+// Load a categorical dataset.
+arma::mat data;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/telecom_churn.arff.
+mlpack::data::Load("telecom_churn.arff", data, info, true);
+
+arma::rowvec responses;
+// See https://datasets.mlpack.org/telecom_churn.responses.csv.
+mlpack::data::Load("telecom_churn.responses.csv", responses, true);
+
+// Split data into training set (80%) and test set (20%).
+arma::mat trainData, testData;
+arma::rowvec trainResponses, testResponses;
+mlpack::data::Split(data, responses, trainData, testData, trainResponses,
+    testResponses, 0.2);
+
+// Create the tree.
+mlpack::AdaBoostRegressor<> adr;
+// Train on the given dataset, specifying a minimum gain of 1e-7 and keeping the
+// default minimum leaf size.
+const double mse = adr.Train(trainData, info, trainResponses, 20 /*number of trees*/
+    ,10 /* minimum leaf size */, 1e-7 /* minimum gain */, 4 /*maximum depth*/);
+
+// Compute test accuracy for each tree
+arma::rowvec testPredictions;
+for(size_t i = 0; i<adr.NumTrees(); i++)
+{
+  adr.Tree(i).Predict(testData, testPredictions);
+  const double testAverageError = arma::mean(testResponses - testPredictions);
+  std::cout << "Tree " << i << " has averageError: " << testAverageError << "."
+    << std::endl;
+}
+
+// Compute predictions on test data using whole forest.
+adr.Predict(testData, testPredictions);
+
+// Compute the average error on the test set.
+const double testAverageError = arma::mean(testResponses - testPredictions);
+std::cout << "The whole forest has Average error: " << testAverageError << "."
```

Comment:
I think maybe the word "model" here is better than "forest"---this isn't a random forest, it's an AdaBoost model.  I know that you could technically call any collection of trees a forest, but I do want to be careful to not accidentally confuse any users.